tests/unit/amo/pages/TestAddon.js

```diff
@@ -750,7 +713,7 @@ describe(__filename, () => {
       description: null,
       summary: null,
     });
-    const root = renderAsDOMNode({ addon });
+    const root = shallowRender({ addon });
```

Comment:
I don't think we should use `shallowRender`, cf. https://github.com/mozilla/addons-frontend/pull/6953#discussion_r234202763

Reply:
Converting all of these away from `shallowRender` is a bigger task, as it's not just changing from `shallowRender` to `renderComponent`, but we also need to make sure the correct data is loaded. As I had to replace `renderAsDOMNode` a number of times, I thought I'd try to keep the diff down by just switching to `shallowRender` for now, and that it all could be cleaned up in #6858.

WDYT?

Reply:
I think we can keep `renderAsDOMNode`. That would reduce the diff and since we need to get rid of `shallowRender` anyway, that's not like it has a huge benefit, no?

Reply:
> `renderAsDOMNode` was creating problems for me with my patch

That could also mean it highlights a bug. Replacing this function by another one to make the test suite pass is a bit dubious, that being said I get the idea.

I'd like to know why this is failing with `renderAsDOMNode` and not `shallowRender` though, because it seems to be a matter of `shallow` vs `mount`.

Reply:
It turns out it was a bug. We were calling `invariant` where it wasn't needed. Thanks for pointing this out. It's fixed now.